### PR TITLE
Converted step completion time to a timestamp from a duration

### PR DESCRIPTION
**Pull request by 0phelia36**
## TL;DR
Converted the time field in Recipe to be a timestamp of when the step will complete instead of a duration of how long the step will execute for. Fixes an issue where if the user refreshes the status page or leaves and comes back mid step the countdown of time remaining is wrong.

## What
What is affected by this PR?
- [X] API
- [X] GUI
- [ ] Hardware Integration
- [ ] OS/Deployment
- [X] Documentation
- [ ] Other (please describe in notes)

## Testing
How was this tested?
- [ ] Locally Virtualized
- [X] Raspberry Pi
- [ ] With Hardware
- [ ] Other (please describe in notes)

## Notes
How was your day?
7/10

### CODE REVIEW

**Review by sparklestheunicorn — APPROVED** [review]

**Review by 0x42red — APPROVED** [review]
Big improvement, lgtm
